runtime/doc/api.txt

```diff
@@ -443,6 +447,41 @@ in the buffer. They could be used to represent cursors, folds, misspelled
 words, and anything else that needs to track a logical location in the buffer
 over time.
 
+Extmarks use |api-indexing|, and you can think of them as cursor positions.
+For example, an extmark at position 1 exists between the first and second
+character on a line, similar to a bar cursor. For that reason, the maximum
+extmark index on a line is 1 larger than the character index: >
+
+     f o o b a r  <-- line contents (with spaces for illustration purposes)
+     0 1 2 3 4 5  <-- character positions (0-based)
+    0 1 2 3 4 5 6 <-- extmark positions (0-based)
+
+Note that extmarks have "right gravity": If you place the cursor directly on
```

Comment:
Would "forward gravity" be a better term?  What happens when `'rightleft'` is set?

Reply:
[Good point](https://github.com/neovim/neovim/pull/12249#issuecomment-671469788) :) I mostly went with the vocabulary I was seeing in https://github.com/neovim/neovim/pull/12249, but I agree something like "forward" would be better.

Reply:
Switched 'right' to 'forward'.